product_category_attributes/data/server_actions.xml

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data noupdate='1'>
+        <record model="ir.actions.server" id="creation_of_product_attributes">
+            <field name="name">Create Product Attributes</field>
+            <field name="model_id" eval="ref('product.model_product_template')"/>
+            <field name="state">code</field>
+            <field name="condition">True</field>
+            <field name="sequence">1</field>
+            <field name="code"><![CDATA[
+categ_obj = pool.get('product.category')
+categ_ids = categ_obj.search(cr, uid, [('id','=',object.categ_id.id)])
+datas = []
+if categ_ids:
+    for categ in categ_obj.browse(cr, uid, categ_ids, context):
+        for attribute in categ.attribute_ids:
+            datas.append((0, 0, {'attribute_id':attribute.id}))
+        object.write({'attribute_line_ids': datas})]]>
+            </field>
+        </record>
+    </data>
+</openerp>
```

Comment:
It is a REALLY cool approach the usage of server actions.

Can you add a little test which trigger this action please, with that I think is complete all the approach and it is not quite difficult due to the on_create.

The test simply,

Create a Category with attribute.
Create a product on that actegory.
Assert the product create has the attribute.

Regards.